owncloudApp/src/main/java/com/owncloud/android/presentation/ui/accounts/AccountManagementActivity.kt

```diff
@@ -0,0 +1,298 @@
+import com.owncloud.android.presentation.ui.authentication.ACTION_UPDATE_TOKEN
+import com.owncloud.android.presentation.ui.authentication.EXTRA_ACTION
+import com.owncloud.android.presentation.ui.authentication.LoginActivity
+import com.owncloud.android.presentation.viewmodels.accounts.AccountsManagementViewModel
+import com.owncloud.android.ui.activity.FileActivity
+import com.owncloud.android.ui.activity.FileDisplayActivity
+import com.owncloud.android.ui.dialog.ConfirmationDialogFragment
+import com.owncloud.android.ui.dialog.RemoveAccountDialogFragment.Companion.newInstance
+import com.owncloud.android.ui.dialog.RemoveAccountDialogViewModel
+import com.owncloud.android.utils.PreferenceUtils
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import timber.log.Timber
+import com.owncloud.android.presentation.ui.authentication.EXTRA_ACCOUNT as EXTRA_ACCOUNT_LOGIN_ACTIVITY
+
+const val KEY_CURRENT_ACCOUNT_CHANGED = "CURRENT_ACCOUNT_CHANGED"
+const val KEY_ACCOUNT_LIST_CHANGED = "ACCOUNT_LIST_CHANGED"
+
+class AccountManagementActivity : FileActivity(), AccountManagementAdapter.AccountAdapterListener, AccountManagerCallback<Boolean> {
+
+    private val accountListAdapter: AccountManagementAdapter = AccountManagementAdapter(this)
+    private lateinit var originalAccounts: Set<String>
+    private lateinit var originalCurrentAccount: String
+    private lateinit var accountBeingRemoved: String
+    private lateinit var tintedCheck: Drawable
+
+    private val removeAccountDialogViewModel: RemoveAccountDialogViewModel by viewModel()
+    private val accountsManagementViewModel: AccountsManagementViewModel by viewModel()
+
+    @Override
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.accounts_layout)
+        tintedCheck = ContextCompat.getDrawable(this, R.drawable.ic_current_white)!!
+        tintedCheck = DrawableCompat.wrap(tintedCheck)
+        val tint = ContextCompat.getColor(this, R.color.actionbar_start_color)
+        DrawableCompat.setTint(tintedCheck, tint)
+
+        val recyclerView: RecyclerView = findViewById(R.id.account_list_recycler_view)
+        recyclerView.run {
+            filterTouchesWhenObscured = PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(applicationContext)
+            adapter = accountListAdapter
+            layoutManager = LinearLayoutManager(this@AccountManagementActivity)
+        }
+
+        setupStandardToolbar(
+            getString(R.string.prefs_manage_accounts),
+            displayHomeAsUpEnabled = true,
+            homeButtonEnabled = true,
+            displayShowTitleEnabled = true
+        )
+
+        val accountList = accountsManagementViewModel.getLoggedAccounts()
+        originalAccounts = toAccountNameSet(accountList)
+        originalCurrentAccount = accountsManagementViewModel.getCurrentAccount()?.name.toString()
+
+
```

Comment:
KtLint may complain if it sees two consecutive blank lines 🤭

Reply:
I didn't have those blank lines, i don't know why you did. In fact I checked KtLint before making the pull request and it didn't complain 😕